packages/workers/src/database/migrations/0003-update-metadata-to-arrays.ts

```diff
@@ -0,0 +1,37 @@
+import type { Pool } from 'pg'
+
+export default async function (client: Pool) {
+  // Add metadata columns to existing logs table
+  // This allows storing multiple metadata values per user (e.g., multiple public keys or token IDs)
+
+  // Add metadata_key and metadata_value columns to logs table
+  await client.query(`
+    ALTER TABLE logs ADD COLUMN IF NOT EXISTS metadata_key TEXT
+  `)
+
+  await client.query(`
+    ALTER TABLE logs ADD COLUMN IF NOT EXISTS metadata_value TEXT
+  `)
+
+  // Add additionalMetadataArguments column to jobs table
+  await client.query(
+    'ALTER TABLE jobs ADD COLUMN IF NOT EXISTS additional_metadata_arguments JSONB',
+  )
+
+  // Add transformUserAddressType column to jobs table
+  await client.query(
+    'ALTER TABLE jobs ADD COLUMN IF NOT EXISTS transform_user_address_type TEXT',
+  )
+
+  // Remove the existing primary key constraint on (address, contract_address)
+  // This allows multiple metadata entries per user-contract pair
+  await client.query(`
+    ALTER TABLE logs DROP CONSTRAINT IF EXISTS logs_pkey
+  `)
```

Comment:
I don't know the exact details of how postgres deals with indexes. I'm pretty certain that creating a primary key would have created an index.

If this is removed, I'd keep a keen eye on how the query, which filters by `address` (user address) performs. If it performs worse than before, you might need to manually add a filter for just that column.